### PR TITLE
Ttt 34 use python logging instead of print statements to give game status information

### PR DESCRIPTION
**Pull request by billlanyon**
Logging is enabled from log.conf, which is set to write debug messages to both the console and to ttt.log. Manual testing has been performed.

### CODE REVIEW

**Review by BasilBibi — APPROVED** [review]
Tis is looking good can you give me demo at AOB?

File: README.md [hunk 1]
```diff
@@ -27,5 +27,7 @@ Players must enter accurate input as requested.
 ### External dependencies
 TBC
 
-### Usage
-TBC
+### Logging
+When launching the app from the command line:<br>
+Appending '-d' enables the display of debug messages in the console.<br>
+Appending '-f' enables the logging of debug messages to a file named 'ttt.log'.
```
Comment (BasilBibi) [inline README.md]: Is this still true ?

Comment (billlanyon) [reply README.md]: Indeed not. I used to think I was relatively good at detail :). The README.md is now updated to reflect reality.

Comment (billlanyon) [reply README.md]: Looking forward to demoing at AOB.

**Comment by billlanyon** [comment]
I have added debug messaging for moves and error messaging with tracebacks when the user inputs non integers. This involved me putting a try/except into process_another_move in main: but this stops the game crashing when this happens, which seemed like a good thing. Here are a couple of screenshots of the logging:
Screenshot 2020-07-23 at 17 11 49
Screenshot 2020-07-23 at 17 52 18

File: com/bl/ttt/tictactoe.py [hunk 2]
```diff
@@ -97,7 +106,7 @@ def process_valid_move(self, move):
         if self._is_computer_game and \
            self.is_computer_turn() and not \
            self.is_game_over():
-            self.get_computer_move()
+            self._get_computer_move()
```
Comment (BasilBibi) [inline com/bl/ttt/tictactoe.py:111]: Might be good to log the state of _player_move_log and move counter at the end of process_valid_move

File: ttt.log [hunk 3]
```diff
@@ -0,0 +1,80 @@
+2020-07-24 15:40:22,570 - com.bl.ttt.main - ERROR - pytest: reading from stdin while output is captured!  Consider using `-s`.
+Traceback (most recent call last):
+  File "/Users/billlanyon/Documents/dev/source/TTT/com/bl/ttt/main.py", line 38, in get_user_input
+    i = input(prompt).upper()
+  File "/opt/anaconda3/lib/python3.7/site-packages/_pytest/capture.py", line 700, in read
```
Comment (BasilBibi) [inline ttt.log]: Looks like you've commited ttt.log by mistake - add an entry in .gitignore we only want source code nothing else in git.

File: com/bl/ttt/tictactoe.py (hunk 2 above)
Comment (billlanyon) [reply com/bl/ttt/tictactoe.py:111]: Done

File: ttt.log (hunk 3 above)
Comment (billlanyon) [reply ttt.log]: ttt.log added to gitignore

**Review by BasilBibi — APPROVED** [review]
Looking good !

**Pull request merged**